ChronoAPI/ChronoLog/include/ChronoLogAdminRPCVisor.h

```diff
@@ -37,50 +41,52 @@ class ChronoLogAdminRPCProxy {
              typeid(*this).name(), this, getpid());
     }
 
-    ~ChronoLogAdminRPCProxy() = default;
+    ~ChronoLogAdminRPCVisor() = default;
 
-    bool LocalConnect(const std::string &uri, std::string &client_id) {
+    int LocalConnect(const std::string &uri, std::string &client_id, int &flags, uint64_t &clock_offset) {
         LOGD("%s in ChronoLogAdminRPCProxy@%p called in PID=%d, with args: uri=%s",
              __FUNCTION__, this, getpid(), uri.c_str());
         ClientRegistryInfo record;
         record.addr_ = "127.0.0.1";
-//        return ChronoLog::Singleton<ClientRegistryManager>::GetInstance()->add_client_record(client_id, record);
-        extern std::shared_ptr<ClientRegistryManager> g_clientRegistryManager;
-        g_clientRegistryManager->add_client_record(client_id, record);
-        return true;
-    }
-
-    bool Connect(const std::string &uri, std::string &client_id) {
-        LOGD("%s in ChronoLogAdminRPCProxy at addresss %p called in PID=%d, with args: uri=%s, client_id=%s",
-             __FUNCTION__, this, getpid(), uri.c_str(), client_id.c_str());
-        return CHRONOLOG_RPC_CALL_WRAPPER("Connect", 0, bool, uri, client_id);
+        if (std::strtol(client_id.c_str(), nullptr, 10) < 0) {
+            LOGE("client id is invalid");
+            return CL_ERR_INVALID_ARG;
+        }
+        if (g_clientRegistryManager) {
+            return g_clientRegistryManager->add_client_record(client_id, record);
+        }
```

Comment:
What do you want to do if the clientRegistryManager does not exist? Do you want to report the error to the server program?

Reply:
I did this just for debugging the segfault I had before. What we can do is to terminate Visor when it is null.

Reply:
ok